File: Stronghold/Stronghold.Application/DTOs/Request/CreateSupplementRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Stronghold.Application.DTOs.Request
{
    public class CreateSupplementRequest
    {
        [Required(ErrorMessage = "Naziv je obavezan.")]
        [StringLength(100, MinimumLength = 2, ErrorMessage = "Naziv suplementa mora imati između 2 i 100 karaktera.")]
        public string Name { get; set; } = string.Empty;

        [Required(ErrorMessage = "Cijena je obavezna.")]
        [Range(0.01, 10000, ErrorMessage = "Cijena mora biti veća od 0 i manja ili jednaka 10000.")]
        public decimal Price { get; set; }

        [StringLength(1000, MinimumLength = 2, ErrorMessage = "Opis suplementa mora imati između 2 i 1000 karaktera.")]
        public string? Description { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Kategorija suplementa je obavezna.")]
        public int SupplementCategoryId { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Dobavljač je obavezan.")]
        public int SupplierId { get; set; }
    }
}

File: packages/stronghold_core/lib/models/requests/create_supplement_request.dart
/// Matches backend CreateSupplementRequest exactly
class CreateSupplementRequest {
  final String name;
  final double price;
  final String? description;
  final int supplementCategoryId;
  final int supplierId;
  final int stockQuantity;

  const CreateSupplementRequest({
    required this.name,
    required this.price,
    this.description,
    required this.supplementCategoryId,
    required this.supplierId,
    this.stockQuantity = 0,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'price': price,
        if (description != null) 'description': description,
        'supplementCategoryId': supplementCategoryId,
        'supplierId': supplierId,
        'stockQuantity': stockQuantity,
      };
}
